app/controllers/cloud_subnet_controller.rb

```diff
@@ -76,42 +76,45 @@ def create
     case params[:button]
     when "cancel"
       javascript_redirect :action    => 'show_list',
-                          :flash_msg => _("Add of new Subnet was cancelled by the user") % {:model => ui_lookup(:table => 'cloud_subnet')}
+                          :flash_msg => _("Add of new Subnet was cancelled by the user") % {
+                            :model => ui_lookup(:table => 'cloud_subnet')}
 
     when "add"
       @subnet = CloudSubnet.new
       options = form_params
       ems = ExtManagementSystem.find(options[:ems_id])
-      valid_action, action_details = CloudSubnet.validate_create_subnet(ems)
-      if valid_action
-        begin
-          CloudSubnet.create_subnet(ems, options)
-          # TODO: To replace with targeted refresh when avail. or either use tasks
-          EmsRefresh.queue_refresh(ManageIQ::Providers::NetworkManager)
-          add_flash(_("Creating %{subnet} \"%{subnet_name}\"") % {
-            :subnet      => ui_lookup(:table => 'cloud_subnet'),
-            :subnet_name => options[:name]})
-        rescue => ex
-          add_flash(_("Unable to create %{subnet} \"%{subnet_name}\": %{details}") % {
-            :subnet      => ui_lookup(:table => 'cloud_subnet'),
-            :subnet_name => options[:name],
-            :details     => ex}, :error)
-        end
-        @breadcrumbs.pop if @breadcrumbs
-        session[:flash_msgs] = @flash_array.dup if @flash_array
-        javascript_redirect :action => "show_list"
+      options.delete(:ems_id)
+      task_id = ems.create_cloud_subnet_queue(session[:userid], options)
+      add_flash(_("Cloud Subnet creation failed: Task start failed: ID [%{id}]") %
+                {:id => task_id.inspect}, :error) unless task_id.kind_of?(Fixnum)
+      if @flash_array
+        javascript_flash(:spinner_off => true)
       else
-        @in_a_form = true
-        add_flash(_(action_details), :error) unless action_details.nil?
-        drop_breadcrumb(
-          :name => _("Add New Subnet") % {:model => ui_lookup(:table => 'cloud_subnet')},
-          :url  => "/cloud_subnet/new"
-        )
-        javascript_flash
+        initiate_wait_for_task(:task_id => task_id, :action => "create_finished")
       end
     end
   end
 
+  def create_finished
+    task_id = session[:async][:params][:task_id]
+    subnet_name = session[:async][:params][:name]
+    task = MiqTask.find(task_id)
+    if MiqTask.status_ok?(task.status)
+      add_flash(_("%{model} \"%{name}\" created") % { :model => ui_lookup(:table => 'cloud_subnet'),
```

Comment:
The `ui_lookup()` here should go away here, we'd rather have "Cloud Subnet" in the string statically (for nicer translations). Same in other places in this PR.